canonical/vanilla-framework
# Add color theme support to Search & Filter component

## Done
Adds color theme support to Search & Filter component by using new color theme variables

- Color of the search and filter panel headers (it is slightly more contrasted from background now, using default text color rather than `#666`)
- Color of the overflow count - it now uses `$colors--theme--link-default` rather than `$color-information` which is unthemed and does not appear nicely on the dark theme
- Background color of opened search and filter panel (now supports dark theme using themed background color instead of hardcoded to light theme)
- Background color of the search prompt (now uses themed alt background color instead of hardcoded to `#f7f7f7`)
- Background color of the search clear button (now uses themed input background color instead of hardcoded whitesmoke/`f5f5f5`)
- Color of search and filter panel section bottom border (now uses low contrast border instead of `mid-x-light`)
- Enabled color theme switcher on search & filter examples
- Drive-bys:
  - Set chip border thickness to input border thickness
  - Adjusted the position of the chip overflow counter to align to the baseline (it was a bit too high).
  - Removes old blue background color of the aside navigation in the JAAS application layout example

Fixes [WD-11868](https://warthogs.atlassian.net/browse/WD-11868)

## QA

- Open all [search and filter examples](https://vanilla-framework-5127.demos.haus/docs/examples/standalone) and verify that the changes above appear as expected in all color themes.

### Check if PR is ready for release

If this PR contains Vanilla SCSS code changes, it should contain the following changes to make sure it's ready for the release:

  - `Feature 🎁`, `Breaking Change 💣`, `Bug 🐛`, `Documentation 📝`, `Maintenance 🔨`.
  - if CSS class names are not changed it can be bugfix relesase (x.x.**X**)
  - if CSS class names are changed/added/removed it should be minor version (x.**X**.0)
  - see the [wiki for more details](https://github.com/canonical/vanilla-framework/wiki/Release-process#pre-release-tasks)


## Screenshots
<img width="1210" alt="Screenshot 2024-08-07 at 10 03 12 AM" src="https://github.com/user-attachments/assets/046e203d-6e51-4f1a-abcd-5decfd67cdd0">
<img width="1114" alt="Screenshot 2024-08-07 at 10 07 19 AM" src="https://github.com/user-attachments/assets/da61d287-38bc-44e7-8f70-42e24aadc8a0">




[WD-11868]: https://warthogs.atlassian.net/browse/WD-11868?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
